scripts/test_jupyter_with_papermill.sh

```diff
@@ -0,0 +1,430 @@
+#! /usr/bin/env bash
+
+## Description:
+## 
+## This script is intended to be invoked via the Makefile test-% target of the notebooks repository and assumes the deploy9-% target
+## has been previously executed.  It replaces the legacy 'test_with_papermill' function previously defined in the Makefile.
+##
+## The script will first check to ensure a notebook workload is running and have a k8s service object exposed.  Once verified:
+##  - the relevant imagestream manifest from https://github.com/opendatahub-io/notebooks/tree/main/manifests/base is copied
+##		into the running pod to act as the "source of truth" when asserting against installed version of py packages
+##  - a test_notebook.ipynb will be copied into the running pod if it is defined in jupyter/*/test/test_notebook.ipynb
+##      - for images inherited from the datascience notebook image, the minimal and datascience notebook test files are
+##          sequentially copied into the running pod
+##  - for each test_notebook.ipynb file that is copied into the running pod, a test suite is invoked via papermill
+##      - test execution is considered failed if the papermill output contains the string 'FAILED'
+##
+## Currently this script only supports jupyter notebooks running on ubi9.
+##
+## Dependencies: 
+##
+##    - git:        https://www.man7.org/linux/man-pages/man1/git.1.html
+##    - kubectl:    https://kubernetes.io/docs/reference/kubectl/
+##      - a local copy of kubectl is downloaded via the Makefile bin/kubectl target, and stored in bin/kubectl within the notebooks repo
+##    - yq:         https://mikefarah.gitbook.io/yq
+##      - a local copy of yq is downloaded via the Makefile bin/yq target, and stored in bin/yq within the notebooks repo
+##    - wget:       https://www.man7.org/linux/man-pages/man1/wget.1.html
+##    - curl:       https://www.man7.org/linux/man-pages/man1/curl.1.html
+##    - kill:       https://www.man7.org/linux/man-pages/man1/kill.1.html
+##
+## Usage: 
+##
+##      test_jupyter_with_papermill.sh <makefile test target>
+##          - Intended to be invoked from the test-% target of the Makefile
+##          - Arguments
+##              - <makefile test target>
+##                  - the resolved wildcard value from the Makefile test-% pattern-matching rule
+##  
+##
+
+
+set -uox pipefail
```

Comment:
```suggestion
-Eeuxo pipefail
```

https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/

Reply:
I had deliberately **not** included `-e` initially...  I have "mixed feelings" on `-e` 's usefulness... as there are caveats/edge cases on when it doesn't actually do anything I still don't have a great handle on..

I am happy to add this in if this is a strong preference - but will then need to do a couple more passes to retest script to make sure do regression in behavior is introduced...

Reply:
We used to have -e in the Makefile. I'm sad to see it go, it feels like an achievement lost (because I added it there ;p) but I don't do much of Bash so I don't have hard Bash preferences.

Reply:
Based on discussions in daily standup - for **NOW** we will proceed without seeing any `-e` / `-E` flag.

This is based on the fact I have not seen the need for it based on my repeated runs of the test script - which seems to have a reasonable outcome in both success and failure cases I have encountered.

Happy to revisit this decision in the future as a specific need/shortcoming arises.